art_tree.go

```diff
@@ -17,40 +20,90 @@ func NewArtTree() *ArtTree {
 	return &ArtTree{root: nil, size: 0}
 }
 
+type Result struct {
+	Key   string
+	Value interface{}
+}
+
+func (t *ArtTree) EachChanResult() chan Result {
+	return t.EachChanResultFrom(t.root)
+}
+
+func (t *ArtTree) EachChanResultFrom(start *ArtNode) chan Result {
+	outChan := make(chan Result)
+	go func() {
+		if start != nil {
+			for n := range t.EachChanFrom(start) {
+				if n.IsLeaf() {
+					outChan <- Result{string(n.key), n.value}
+				}
+			}
+		}
+		close(outChan)
+	}()
+	return outChan
+}
```

Comment:
Could you elaborate a bit more on the desire to have a channel based approach here?  It seems to me that this code is returning a channel while it asynchronously processes the art tree and appends values to the outChan.  

Could the returned channel ever have lock contention?  What if I wanted to use it immediately after calling `EachChanResultFrom`?

Would this method be easier to use as a client if a `[]Result` was returned instead?